File: Java/src/com/paracamplus/ilp1/ilp1tme3/C/sinus.c
#include "sinus.h"
#include "ilp.h"
#include <stdio.h>
#include <stdlib.h>
#include <stdarg.h>
#include <string.h>
#include <setjmp.h>
#include <math.h>

ILP_Object
ILP_sinus (ILP_Object val)
{
     if ( ILP_isInteger(val) ) {

     return ILP_make_float(sin(val));

     }else if ( ILP_isFloat(val) ) {

    	 return ILP_make_float(sin(val));
    }
     return  ILP_domain_error("une valeur non numérique", val);
}

File: Java/src/com/paracamplus/ilp1/ilp1tme3/interpreter/primitive/Sinus.java
/* *****************************************************************
 * ILP9 - Implantation d'un langage de programmation.
 * by Christian.Queinnec@paracamplus.com
 * See http://mooc.paracamplus.com/ilp9
 * GPL version 3
 ***************************************************************** */
package com.paracamplus.ilp1.ilp1tme3.interpreter.primitive;


import java.math.BigInteger;
import com.paracamplus.ilp1.interpreter.interfaces.EvaluationException;
import com.paracamplus.ilp1.interpreter.primitive.UnaryPrimitive;

public class Sinus extends UnaryPrimitive {
    
    public Sinus() {
        super("sinus");
    }
        
    @Override
	public Object apply (Object value) throws EvaluationException {      
           if(value instanceof Double) {
        	   return Math.sin((Double)value);
           }
           else if(value instanceof BigInteger) {
        	   return Math.sin(((BigInteger)value).doubleValue());
           } 
           else throw new EvaluationException("une valeur non numérique");

    }
}
